src/servant/tt_002/faerie.c

```diff
@@ -1983,7 +1980,11 @@ void func_us_80177F84(Entity* self) {
 
     posX2 = self->posX.i.hi;
     posY = self->posY.i.hi;
-    unkStruct = &D_us_801724CC[LOH(self->params)];
+    // The data in D_us_801724CC is aligned in a way that it needs to be a u16
+    // array So this is a bit ugly to cast it into the struct pointer
+    unkStruct =
```

Comment:
This seems odd to me, was it an array originally?

Reply:
D_us_801724CC Is likely an array and we’re pulling the offset based on param. Ideally, we would use an array of that type, but whenever I would use that type, 2 extra bytes would get padded in to each member.  Since that struct has both an u16 and a s32 in addition to s16, I’m pretty sure it originally was a struct, or they were doing some really weird pointer math.